client/src/main/java/org/opensearch/ml/client/MachineLearningClient.java

```diff
@@ -337,4 +339,22 @@ default ActionFuture<ToolMetadata> getTool(String toolName) {
      * @param listener action listener
      */
     void getTool(String toolName, ActionListener<ToolMetadata> listener);
+
+    /**
+     * Registers new agent and returns ActionFuture.
+     * @param mlRegisterAgentRequest Register agent input, refer: https://opensearch.org/docs/latest/ml-commons-plugin/api/#register-agent
+     * @return the result future
+     */
+    default ActionFuture<MLRegisterAgentResponse> registerAgent(MLRegisterAgentRequest mlRegisterAgentRequest) {
```

Comment:
Based on the previous patterns of the APIs, we generally create the request in `MachineLearningNodeClient`
```suggestion
    default ActionFuture<MLRegisterAgentResponse> registerAgent(MLAgent mlAgent) {
```